scanner/updater/manual/manual.go

```diff
@@ -1,39 +1,220 @@
-// Package manual provides a custom updater for vulnerability scanner.
+// Package manual provides a custom updater for vulnerability scanners.
 // This updater allows manual input of vulnerability data.
 package manual
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
 
+	"github.com/pkg/errors"
 	"github.com/quay/claircore"
 	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/zlog"
+	"github.com/stackrox/rox/pkg/httputil/proxy"
+	"github.com/stackrox/rox/pkg/utils"
 )
 
-// Name provides the name for the updater.
-const Name = `stackrox-manual`
+type Vulnerability struct {
+	Name               string `json:"Name"`
+	Description        string `json:"Description"`
+	Issued             string `json:"Issued"`
+	Links              string `json:"Links"`
+	Severity           string `json:"Severity"`
+	NormalizedSeverity string `json:"NormalizedSeverity"`
+	Package            struct {
+		Name           string `json:"Name"`
+		RepositoryHint string `json:"RepositoryHint"`
+	} `json:"Package"`
+	IntroducedInVersion string `json:"IntroducedInVersion"`
+	FixedInVersion      string `json:"FixedInVersion"`
+	Repo                struct {
+		Name string `json:"Name"`
+		URI  string `json:"URI"`
+	} `json:"Repo"`
+}
 
-// UpdaterSet creates a new updater set with the provided vulnerability data.
-func UpdaterSet(_ context.Context) (driver.UpdaterSet, error) {
-	res := driver.NewUpdaterSet()
-	err := res.Add(&updater{})
+type Vulnerabilities struct {
+	Vulnerabilities []Vulnerability `json:"vulnerabilities"`
+}
+
+const (
+	Name = "stackrox-manual"
+	// DefaultURL Default URL to fetch the vulnerabilities JSON.
+	DefaultURL = "https://raw.githubusercontent.com/stackrox/stackrox/master/scanner/updater/manual/vulns.json"
+)
+
+var defaultClient = &http.Client{
+	Transport: proxy.RoundTripper(),
+	Timeout:   5 * time.Minute,
+}
+
+type updater struct {
+	c         *http.Client
+	updateURL *url.URL
+}
+
+// NewUpdater creates a new instance of the updater with default settings.
+func NewUpdater(c *http.Client, uri string) (*updater, error) {
+	var url *url.URL
+	var err error
+	if uri == "" {
+		uri = DefaultURL
+	}
+	url, err = url.Parse(uri)
 	if err != nil {
-		return res, err
+		return nil, err
 	}
-	return res, nil
+	return &updater{c: c, updateURL: url}, nil
+}
+
+func (u *updater) Name() string {
+	return Name
+}
+
+// Fetch fetching data from a configurable URI.
+func (u *updater) Fetch(ctx context.Context, fingerprint driver.Fingerprint) (io.ReadCloser, driver.Fingerprint, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "updater/manual/manual.Fetch")
+	resp, err := u.c.Get(u.updateURL.String())
```

Comment:
This is probably ok for this simple task outside of the product itself, but usually we want to pass a context into the request